### PR TITLE
Spring boot starters for API and Gateway services

### PR DESCRIPTION
**Pull request by paulcwarren**
Probably, overly simple but putting it out there for discussion.

### CODE REVIEW

File: thunx-api-spring-boot-starter/src/main/resources/spring.provides [hunk 1]
```diff
@@ -0,0 +1 @@
+provides: thunx-spring
```
Comment (tgeens) [inline thunx-api-spring-boot-starter/src/main/resources/spring.provides:1]: Spring Boot itself removed all spring.provides files in 2018
See https://github.com/spring-projects/spring-boot/issues/12435 for more information

File: thunx-api-spring-boot-starter/src/main/java/eu/xenit/contentcloud/thunx/api/autoconfigure/AbacAutoConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package eu.xenit.contentcloud.thunx.api.autoconfigure;
```
Comment (tgeens) [inline thunx-api-spring-boot-starter/src/main/java/eu/xenit/contentcloud/thunx/api/autoconfigure/AbacAutoConfiguration.java:1]: I think this is an erroneous duplicate ?

**Review by tgeens — APPROVED** [review]

**Review by tgeens — APPROVED** [review]

File: thunx-autoconfigure/src/main/java/eu/xenit/contentcloud/thunx/gateway/autoconfigure/GatewayAutoConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        return OpaClient.builder()
+                .httpLogging(RestClientConfiguration.LogSpecification::all)
+                .url(opaProperties.getUrl())
+                .build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public PolicyDecisionPointClient pdpClient(OpaClient opaClient) {
+        return new OpenPolicyAgentPDPClient(opaClient, request -> "data.vfinance.brokercontent.allow == true");
```
Comment (vierbergenlars) [inline thunx-autoconfigure/src/main/java/eu/xenit/contentcloud/thunx/gateway/autoconfigure/GatewayAutoConfiguration.java:36]: The OPA query here is very specific.
It should either be configurable with some property, or else this bean can't really be autoconfigured at all.

Comment (tgeens) [reply thunx-autoconfigure/src/main/java/eu/xenit/contentcloud/thunx/gateway/autoconfigure/GatewayAutoConfiguration.java:36]: True, lets fix that by introducing the `OpaQueryProvider` as a dependency :+1:

File: thunx-autoconfigure/build.gradle [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    compileOnly 'org.springframework.cloud:spring-cloud-starter-gateway'
+
+    annotationProcessor "org.springframework.boot:spring-boot-autoconfigure-processor:${springBootBomVersion}"
+    annotationProcessor "org.projectlombok:lombok:${lombokVersion}"
+
+    testImplementation platform("org.springframework.boot:spring-boot-dependencies:${springBootBomVersion}")
+    testImplementation platform("org.springframework.cloud:spring-cloud-dependencies:${springCloudVersion}")
+
+    testImplementation 'org.springframework.boot:spring-boot-starter-test'
+    testImplementation 'org.assertj:assertj-core:3.21.0'
```
Comment (tgeens) [inline thunx-autoconfigure/build.gradle:32]: These versions should be managed by the spring-boot platform ?

File: thunx-autoconfigure/build.gradle [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    compileOnly platform("org.springframework.boot:spring-boot-dependencies:${springBootBomVersion}")
+    compileOnly platform("org.springframework.cloud:spring-cloud-dependencies:${springCloudVersion}")
+
+    implementation "org.springframework.boot:spring-boot-autoconfigure"
+
+    compileOnly "org.projectlombok:lombok"
+
+    compileOnly project(':thunx-spring')
+    compileOnly project(':thunx-pdp-opa')
+    compileOnly "eu.xenit.contentcloud:opa-async-java-client:0.2.0"
```
Comment (tgeens) [inline thunx-autoconfigure/build.gradle:18]: can we externalize these versions ?

**Review by tgeens — APPROVED** [review]
Minor feedback about dependency versions can be addressed in the next PR.

**Pull request merged**